### PR TITLE
KAFKA-17759: Remove Utils.mkSet

### PR DESCRIPTION
**Pull request by gaurav-narula**
Removes `Utils#mkSet` as JDK 9+ allows `Set#of` for creating a set.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by gaurav-narula** [comment]
CC: @chia7712

**Review by mumrah — COMMENTED** [review]
@gaurav-narula thanks for the patch! 

On significant difference is that `Set#of` returns an immutable set. Is it acceptable for all of our usages to use an immutable set?

**Comment by gaurav-narula** [comment]
Thanks for your review @mumrah 

> On significant difference is that `Set#of` returns an immutable set. Is it acceptable for all of our usages to use an immutable set?

Indeed. Almost all of the usages of `mkSet` are in test files, where we usually don't mutate the set. In some cases, where a mutable Set is required, I substituted the declaration with `new HashSet<>(List.of(...))`

I could find the following usages of `mkSet` in production code and here are some observations:

1. `SslConfigs#RECONFIGURABLE_CONFIGS` and `SslConfigs#NON_RECONFIGURABLE_CONFIGS`. There's a getter for the former via the `Reconfigurable` interface. I don't think the interface requires the set to be mutable as SocketServer.scala implements the same interface and uses a Scala Set which is immutable by default.
2. Topic#INTERNAL_TOPICS where we eventually wrap it in `Collections.unmodifiableSet`
3. JmxReporter#RECONFIGURABLE_CONFIGS - similar to (1)
4. ScramSaslServer#SUPPORTED_EXTENSIONS - couldn't find any mutation operations on the set.
5. JsonConverter - passed into JsonSerializer/JsonDeserializer. Couldn't find any mutation operations on it.
6. StandardAuthorizerData - couldn't find any mutation operations on the set returned
7. LogConfig - wrapped in `Collections.unmodifiableSet`
8. ClusterInstance#supportedGroupProtocols() - only used in tests

File: clients/src/main/java/org/apache/kafka/common/internals/Topic.java [hunk 1]
```diff
@@ -37,7 +36,7 @@ public class Topic {
     public static final String LEGAL_CHARS = "[a-zA-Z0-9._-]";
 
     private static final Set<String> INTERNAL_TOPICS = Collections.unmodifiableSet(
```
Comment (chia7712) [inline clients/src/main/java/org/apache/kafka/common/internals/Topic.java]: The doc of `Set.of` says it return unmodifiable set, so we don't need to wrap it again, right?

File: storage/src/main/java/org/apache/kafka/storage/internals/log/LogConfig.java [hunk 2]
```diff
@@ -205,7 +205,7 @@ public Optional<String> serverConfigName(String configName) {
     private static final String MESSAGE_TIMESTAMP_DIFFERENCE_MAX_MS_DOC = TopicConfig.MESSAGE_TIMESTAMP_DIFFERENCE_MAX_MS_DOC;
 
     // Visible for testing
-    public static final Set<String> CONFIGS_WITH_NO_SERVER_DEFAULTS = Collections.unmodifiableSet(Utils.mkSet(
+    public static final Set<String> CONFIGS_WITH_NO_SERVER_DEFAULTS = Collections.unmodifiableSet(Set.of(
```
Comment (chia7712) [inline storage/src/main/java/org/apache/kafka/storage/internals/log/LogConfig.java]: ditto. we don't need redundant wrap

File: metadata/src/test/java/org/apache/kafka/controller/ReplicationControlManagerTest.java [hunk 3]
```diff
@@ -491,7 +491,7 @@ void alterTopicConfig(
         }
 
         void fenceBrokers(Integer... brokerIds) {
-            fenceBrokers(Utils.mkSet(brokerIds));
+            fenceBrokers(Set.of(brokerIds));
```
Comment (chia7712) [inline metadata/src/test/java/org/apache/kafka/controller/ReplicationControlManagerTest.java:486]: The failure is caused by the undefined order. I file https://issues.apache.org/jira/browse/KAFKA-17772 to fix it for now.

**Review by chia7712 — COMMENTED** [review]
@gaurav-narula thanks for this patch

**Comment by chia7712** [comment]
@gaurav-narula Could you please fix the conflicts and import errors?

**Comment by gaurav-narula** [comment]
@chia7712 Please have a look now

**Pull request merged**